### PR TITLE
allow alternative variants per server

### PR DESCRIPTION
**Pull request by deads2k**
This will allow us to specify different variants for kube.  It just makes an instance variable that is passed around.

### CODE REVIEW

File: pkg/testgridanalysis/testidentification/variants.go [hunk 1]
```diff
@@ -1,156 +1,16 @@
...
 
-	"github.com/openshift/sippy/pkg/util/sets"
-	"k8s.io/klog"
-)
+// VariantManager identifies and describes different variants
+type VariantManager interface {
+	// allOpenshiftVariants returns a set of all known variants
+	AllVariants() sets.String
 
-var (
```
Comment (deads2k) [inline pkg/testgridanalysis/testidentification/variants.go:11]: this all moved to ocp_variants

File: pkg/testgridanalysis/testidentification/ocp_variants.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/openshift/sippy/pkg/util/sets"
+	"k8s.io/klog"
+)
+
+var (
```
Comment (deads2k) [inline pkg/testgridanalysis/testidentification/ocp_variants.go:11]: a file move and interface alignment.

File: main.go [hunk 3]
```diff
@@ -190,18 +191,38 @@ func (o *Options) runCLIReportMode() error {
...
+	if o.SkipBugLookup || !o.hasOpenshiftRelease() {
+		return buganalysis.NewNoOpBugCache()
+	} else {
+		return buganalysis.NewBugCache()
+	}
+}
+
+func (o *Options) getVariantManager() testidentification.VariantManager {
+	if o.hasOpenshiftRelease() {
+		return testidentification.NewOpenshiftVariantManager()
```
Comment (bparees) [inline main.go:220]: it's not immediately obvious to me why the set of variants should be tied to whether or not what we are analyzing includes an openshift release version?

Comment (deads2k) [reply main.go:220]: > it's not immediately obvious to me why the set of variants should be tied to whether or not what we are analyzing includes an openshift release version?

Depending on what test-grid you're looking at, you have different variants.  See https://testgrid.k8s.io/sig-release-master-informing for instance.  It's likely to have a kube set

Comment (bparees) [reply main.go:220]: yeah, i get that you might have a reason to want different variants depending what set of testgrid jobs you are analyzing.  i'm just not convinced "we have an openshift release" is the right discriminator for choosing the appropriate set of variants.

Comment (deads2k) [reply main.go:220]: > yeah, i get that you might have a reason to want different variants depending what set of testgrid jobs you are analyzing. i'm just not convinced "we have an openshift release" is the right discriminator for choosing the appropriate set of variants.

So this is trying to determine: "am I looking at a set of testgrids that includes an openshift".  How would you like to determine that?

**Comment by bparees** [comment]
/lgtm

with iou on "add a --variant-manager" option and do something more explicit in terms of defaulting the variant manager rather than the currently somewhat fragile/limited "do we have an openshift release value"(which doesn't work if someone doesn't specify --release) decision logic.

**Pull request merged**